### DISCUSSION TITLE
GitHub Copilot could not connect to server.

### DISCUSSION THREAD

**Discussion by yadgen** [community/community#7707] [Copilot Conversations] [answered]
GitHub Copilot could not connect to server. Extension activation failed: "User not authorized".
How can I fix this problem? thanks.

**Comment by davecheney** [comment, marked as answer]
We’re adding people as we have capacity, thanks for your patience and understanding.

**Reply by Mohammed-Shakeeb-ML** [reply]
i too have the same issues will i get authentication for this

**Reply by LouisIV** [reply]
If you had the missing token error like I did after a few days of using CP. I fixed it by signing out of GitHub in VSCode. Uninstalling CP. Then signing back into GitHub and re-installing CP. Really annoying though

**Reply by manuel231509** [reply]
How can I log out of github in VSCode, which is CP, would it be COPILOT? I also have that error, from GitHub Copilot

**Reply by sodiqjonovich** [reply]
i have problem such as it. Then i supposed "I will uninstall and re-install it" . But this was't useful for me. Can you help me

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by Mohammed-Shakeeb-ML** [reply]
Does anyone got the resolution for the above issue

**Reply by amadich** [reply]
how?

**Reply by ssmith777** [reply]
This is the error i receive when attempting to use the product.  [2021-11-24T02:34:40.735Z] GitHub Copilot could not connect to server. Extension activation failed: "undefined" @davecheney please advise?

**Reply by 7-happy** [reply]
it seems that  the "Technical Preview" version has a waitlist, we should join in waitlist and wait for the email.

**Reply by goreprasad1989** [reply]
I logged in from a different account while the signup process by mistake .....so now it shows this message "GitHub Copilot could not connect to server. Extension activation failed: "User not authorized", now how to authorize or sign in from correct account on vscode, I am using mac by the way

**Reply by 7-happy** [reply]
> I logged in from a different account while the signup process by mistake .....so now it shows this message "GitHub Copilot could not connect to server. Extension activation failed: "User not authorized", now how to authorize or sign in from correct account on vscode, I am using mac by the way

i got the same mistake

**Reply by davecheney** [reply]
@7-happy i'm sorry, you are on the waitlist. We are inviting users as we have capacity. Thank you for your patience and your understanding.

**Reply by cin181920** [reply]
How many day we can invite in waitlist on github copilot ? @davecheney

**Reply by Kyivnik** [reply]
I have the same problem. I'm already waiting on the waiting list for 3-4 months, I really want to try this addition and write an article :(

**Reply by J-A-V-Y-21** [reply]
I have the same problem. I'm already waiting on the waiting list for 2 months. :'(

**Reply by rtaparay** [reply]
recien me acabo de suscribir cuantos meses son?

**Reply by sumariyanto** [reply]
same, i have like this problem, my autopilot not working, any answer?

**Reply by arafatx** [reply]
Really need to test this out. I believe this error means I'm still on waiting list to try out GitHub Copilot:

```
[INFO] [auth] [2021-12-13T03:46:48.903Z] Invalid copilot token: missing token: 404 
[ERROR] [default] [2021-12-13T03:46:48.910Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

**Reply by gildomingos** [reply]
**Olá, estou tentando usar o copilot más diz essa mensagem de erro ou seja eu preciso de alguma permissão do desenvolvedor para eu conseguir usar ?**
GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

**Comment by StarLederer** [comment]
Are we supposed to be getting these errors even thought we are added to the preview program?

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Comment by MohamedElashri** [comment]
I'm getting this error also, although I'm being in preview program for a while now.

**Reply by shuyinzjk** [reply]
I'm having the same trouble even though I'm already in the technical preview program. What should I do next? @davecheney 
<img width="449" alt="image" src="https://user-images.githubusercontent.com/66748083/144122418-78ebd6a7-5e5b-4fcc-a5c0-7802b7b3e09b.png">

**Reply by davecheney** [reply]
Looks like a weird Javascript bug, try upgrading vscode and copilot.

**Reply by gj1118** [reply]
@davecheney I am on the latest version of vscode and the latest version of copilot and yet I am receiving the same message ..  _I am already in the technical preview program._ Strangely enough it was working for me a few days ago. That said, following is the error I am getting in the console. 

![image](https://user-images.githubusercontent.com/473101/144353635-2b4b31c8-241c-43f9-b4ce-a41f7f00422c.png)

I am on the following version of Copilot.

![image](https://user-images.githubusercontent.com/473101/144353676-5a182af5-9ac7-44e5-b935-7fa58441ce3b.png)

The following is the VsCode version information.

```
Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.1.0
```

I am on MacOS version 12.0.1.

Please note that I am on a company laptop and there are _some self signed certificates_ .  But do please note that I have duly marked them as trusted .  I have also tried logging out and logging back into Github multiple times from Github in Vscode. It does not help either. I have also tried restarting vscode between disabling and enabling the extension, but even then it does not work .. 

I also tried pinging `api.github.com` and it seems that its able to find it .. please see the ping result below..
![image](https://user-images.githubusercontent.com/473101/144355810-6423da94-ae6f-4e58-b07e-4e1b1a16548a.png)


Please suggest if there is anything else that remains to be tried :) 

Thank you ..

**Reply by gj1118** [reply]
@shuyinzjk  were you able to resolve this issue ? @davecheney can you please help me out with the issue mentioned above please ? 

thank you so much for your time and patience.

**Reply by davecheney** [reply]
@gj1118 I'm sorry you're having problems. The best advice I can offer is this is an issue with the computer you are using or the network you are on. Copilot does not work behind firewalls that intercept SSL traffic and does not currently support http proxies. If your network requires the use of those technologies, this is likely to be the problem.

**Reply by gj1118** [reply]
No worries @davecheney .. I had somehow realized that as well because the message was clear enough .. Its nice to know that currently it does not support proxies .. Back to writing code again ;) 

thanks for all the help though.. much appreciated ..

thanks..

**Comment by MincDev** [comment]
Me too. I've been waiting to be added for a few months already. One of my team members got access to the preview program within a week of joining the waitlist. Not sure how this works.

**Comment by davecheney** [comment]
This error indicates that you are either not part of the beta invite list, or there was an error connecting to api.github.com to request a token to use the Copilot service. Can you please past the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by yadgen** [reply]
> This error indicates that you are either not part of the beta invite list, or there was an error connecting to api.github.com to request a token to use the Copilot service. Can you please past the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

`[INFO] [auth] [2021-11-16T03:10:55.638Z] Invalid copilot token: missing token`

**Reply by yadgen** [reply]
I have authorized this.

**Reply by davecheney** [reply]
This looks like an installation error. Please uninstall copilot and follow the instructions documented here https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/gettingstarted.md#getting-started-with-github-copilot-in-visual-studio-code

**Reply by yadgen** [reply]
> This looks like an installation error. Please uninstall copilot and follow the instructions documented here https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/gettingstarted.md#getting-started-with-github-copilot-in-visual-studio-code

`It didn't work, and it didn't help me. I have installed it many times from yesterday to today.`

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by yadgen** [reply]
> Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

`Hello, thank you very much for your reply.
I just tried it many times, and it didn’t work if I turned off the VPN or turned on the VPN.
After Click Signing in to GitHub, Report an errro GitHub Copilot could not connect to server. Extension activation failed: "User not authorized".`

**Reply by davecheney** [reply]
Something on your machine or your local network appears to be blocking access to api.github.com. Sorry, that's all I can suggest.

**Reply by yadgen** [reply]
> Something on your machine or your local network appears to be blocking access to api.github.com. Sorry, that's all I can suggest.

Please check:
<img width="521" alt="截屏2021-11-16 下午1 10 23" src="https://user-images.githubusercontent.com/1206012/141924728-8654316f-b493-4b05-ad89-e312e69cb2e1.png">

**Reply by Ophuscado** [reply]
Ensure that you are using the original Visual Studio Code build from Microsoft. Currently, the Github Copilot extension doesn't work with VSCodium. The install instructions could have a warning because it's a hard to debug situation.

**Reply by yadgen** [reply]
> Ensure that you are using the original Visual Studio Code build from Microsoft. Currently, the Github Copilot extension doesn't work with VSCodium. The install instructions could have a warning because it's a situation hard to debug.

This is my version.
Version: 1.62.3
Submit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z (1 day ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.1.0

**Reply by NutMaulana4554** [reply]
so how to solve it i don't know what i should do

**Reply by davecheney** [reply]
@NutMaulana4554 i'm sorry, you are on the waitlist. We are inviting users as we have capacity. Thank you for your patience and your understanding.

**Comment by machir** [comment]
I am facing the same issue for several days!!!
no solution till now :(

**Comment by Nymph2333** [comment]
I am facing the same issue for several days too :(

**Comment by yadie-ray** [comment]
![copilot error](https://user-images.githubusercontent.com/71276704/143686650-a727049e-0fb6-4960-b141-a78f7e32f474.jpg)
![ping api github](https://user-images.githubusercontent.com/71276704/143686829-334e57ec-bd27-4066-8a2a-013d288b65d6.jpg)

yeah, me too.. I am facing the same issue and still no solution :(

**Comment by Sid110307** [comment]
Is there any solution for this yet?

**Reply by yadie-ray** [reply]
not yet, some friends who use this suggest me to wait for an email notification if the waitlist has been approved. some are fast but some are slow to receive the notification email, some even take about three months :(

**Reply by Sid110307** [reply]
I already am part of the technical preview. It works with VSCode on my Windows machine, but not on my Linux machine.

**Reply by Sid110307** [reply]
This is the output log:
```log
[INFO] [auth] [...] Invalid copilot token: missing token: 404 
[ERROR] [default] [...] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

**Reply by davecheney** [reply]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by Sid110307** [reply]
I did that, it asked me to sign in, after that it still says "User not authorized".
I tried removing the extension folder in ~/.vscode/extensions, but it still doesn't work. I am already part of the technical preview

**Comment by alexbunardzic** [comment]
I have successfully installed Copilot on one machine, but when attempting to install it on another machine, it failed because I was logged into GitHub with a different account. I then logged out and logged in with my regular GitHub account, but I cannot get Copilot to authenticate. It's almost like it's holding on to the old account and insists that it cannot authenticate.

How to fix that?

**Reply by davecheney** [reply]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Comment by barisdevjs** [comment]
I have also this problem, too by tonight. So any solutions from anyone?

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by barisdevjs** [reply]
First of all, It works properly now. FIXED
When I first opened my PC without Wifi I see a GitHub co-pilot error. After a short time I connected to Wifi, but It was just the same. So I coded away about 4-5 hours. When I saw your message, I restarted VsCode and now It works fine.
Thanks for your attention

**Reply by mebsahle** [reply]
Thanks @barisdevjs .
It's working now.

**Comment by yadgen** [comment]
@davecheney 
Can you provide it for me? thank you very much.
<img width="641" alt="截屏2021-12-03 上午11 05 21" src="https://user-images.githubusercontent.com/1206012/144538107-8ad68e57-ba8d-4f6a-843c-f89da1c5f03e.png">

**Reply by Sid110307** [reply]
[#1644339](https://github.com/github/feedback/discussions/7707#discussioncomment-1644339)

**Reply by davecheney** [reply]
@yadgen i'm sorry, you are on the waitlist. We are inviting users as we have capacity. Thank you for your patience and your understanding.

**Comment by JackCloudman** [comment]
I had the same problem in WSL2, just changed the file **/etc/resolv.conf** with other nameserver

```
nameserver 8.8.8.8
```

**Comment by aidantorrence** [comment]
I fixed this by uninstalling and installing copilot a couple times until i got a prompt to re-authorize
